homeassistant/components/switch/insteon_plm.py

```diff
@@ -56,42 +65,119 @@ def should_poll(self):
     @property
     def address(self):
         """Return the address of the node."""
-        return self._address
+        return self._device.address.human
 
     @property
-    def name(self):
+    def id(self):
         """Return the name of the node."""
-        return self._name
+        return self._device.id
+
+    @property
+    def name(self):
+        """Return the name of the node. (used for Entity_ID)"""
+        if self._newnames:
+            return self._device.id + '_' + self._state.name
+        else:
+            if self._state.group == 0x01:
+                return self._device.id
+            else:
+                return self._device.id+'_'+str(self._state.group)
 
     @property
     def is_on(self):
         """Return the boolean response if the node is on."""
-        onlevel = self._plm.get_device_attr(self._address, 'onlevel')
-        _LOGGER.debug('on level for %s is %s', self._address, onlevel)
+        onlevel = self._state.value
+        _LOGGER.debug('on level for %s is %s', self._device.id, onlevel)
         return bool(onlevel)
 
     @property
     def device_state_attributes(self):
         """Provide attributes for display on device card."""
         insteon_plm = get_component('insteon_plm')
-        return insteon_plm.common_attributes(self)
+        return insteon_plm.common_attributes(self._device, self._state)
```

Comment:
We probably don't want to log the common attributes

Reply:
Thank you again for all your time. Per the comment above, I am not clear on this recommendation.